common/network-shuffle/src/main/java/org/apache/spark/network/shuffle/RemoteBlockPushResolver.java

```diff
@@ -1020,6 +1019,81 @@ private AppPathsInfo(
     }
   }
 
+  /** Merged Shuffle related information tracked for a specific application attempt */
+  public static class AppShuffleInfo {
+
+    private final String appId;
+    private final int attemptId;
+    private final AppPathsInfo appPathsInfo;
+    private final ConcurrentMap<Integer, Map<Integer, AppShufflePartitionInfo>> partitions;
+
+    AppShuffleInfo(
+      String appId,
+      int attemptId,
+      AppPathsInfo appPathsInfo
+    ) {
+      this.appId = appId;
+      this.attemptId = attemptId;
+      this.appPathsInfo = appPathsInfo;
+      partitions = Maps.newConcurrentMap();
+    }
+
+    /**
+     * The logic here is consistent with
+     * @see [[org.apache.spark.storage.DiskBlockManager#getMergedShuffleFile(
+     *      org.apache.spark.storage.BlockId, scala.Option)]]
+     */
+    private File getFile(String filename) {
+      // TODO: [SPARK-33236] Change the message when this service is able to handle NM restart
+      File targetFile = ExecutorDiskUtils.getFile(appPathsInfo.activeLocalDirs,
+        appPathsInfo.subDirsPerLocalDir, filename);
+      logger.debug("Get merged file {}", targetFile.getAbsolutePath());
+      return targetFile;
+    }
+
+    private String generateFileName(
+      String appId,
+      int shuffleId,
+      int reduceId) {
+      return String.format(
+        "%s_%s_%d_%d", MERGED_SHUFFLE_FILE_NAME_PREFIX, appId, shuffleId, reduceId);
+    }
+
+    public File getMergedShuffleDataFile(
+        int shuffleId,
+        int reduceId) {
+      String fileName = String.format("%s.data", generateFileName(appId, shuffleId, reduceId));
+      return getFile(fileName);
+    }
+
+    public File getMergedShuffleIndexFile(
+        int shuffleId,
+        int reduceId) {
+      String indexName = String.format("%s.index", generateFileName(appId, shuffleId, reduceId));
+      return getFile(indexName);
+    }
+
+    public File getMergedShuffleMetaFile(
+      int shuffleId,
+      int reduceId) {
+      String metaName = String.format("%s.meta", generateFileName(appId, shuffleId, reduceId));
+      return getFile(metaName);
+    }
+
+    /**
+     * Clean up the AppShufflePartitionInfo for a specific AppShuffleInfo.
+     */
+    public void cleanupShufflePartitionInfo() {
```

Comment:
Nit: This is closing all the shuffle files of all the partitions belonging to application. So can we  call it `closeAllShufflePartitions`?